src/Common/test/Certificates.Test/ConfigureCertificateOptionsTest.cs

```diff
@@ -17,176 +18,287 @@ public sealed class ConfigureCertificateOptionsTest
+    public async Task CertificateOptionsUpdateOnFileLocationChange(string certificateName)
     {
         using var sandbox = new Sandbox();
-        string instance1Certificate = await File.ReadAllTextAsync("instance.crt", TestContext.Current.CancellationToken);
-        string instance1PrivateKey = await File.ReadAllTextAsync("instance.key", TestContext.Current.CancellationToken);
-        var firstX509 = X509Certificate2.CreateFromPemFile("instance.crt", "instance.key");
-        string instance2Certificate = await File.ReadAllTextAsync("instance2.crt", TestContext.Current.CancellationToken);
-        string instance2PrivateKey = await File.ReadAllTextAsync("instance2.key", TestContext.Current.CancellationToken);
-        var secondX509 = X509Certificate2.CreateFromPemFile("instance.crt", "instance.key");
-        string certificate1FilePath = sandbox.CreateFile("cert", instance1Certificate);
-        string privateKey1FilePath = sandbox.CreateFile("key", instance1PrivateKey);
-        string certificate2FilePath = sandbox.CreateFile("cert2", instance2Certificate);
-        string privateKey2FilePath = sandbox.CreateFile("key2", instance2PrivateKey);
+        string firstInstanceCertificate = await File.ReadAllTextAsync("instance.crt", TestContext.Current.CancellationToken);
+        string firstInstancePrivateKey = await File.ReadAllTextAsync("instance.key", TestContext.Current.CancellationToken);
+        using var firstX509 = X509Certificate2.CreateFromPemFile("instance.crt", "instance.key");
+        string firstCertificateFilePath = sandbox.CreateFile(Guid.NewGuid() + ".crt", firstInstanceCertificate);
+        string firstPrivateKeyFilePath = sandbox.CreateFile(Guid.NewGuid() + ".key", firstInstancePrivateKey);
+        string secondInstanceCertificate = await File.ReadAllTextAsync("secondInstance.crt", TestContext.Current.CancellationToken);
+        string secondInstancePrivateKey = await File.ReadAllTextAsync("secondInstance.key", TestContext.Current.CancellationToken);
+        using var secondX509 = X509Certificate2.CreateFromPemFile("secondInstance.crt", "secondInstance.key");
+        string secondCertificateFilePath = sandbox.CreateFile(Guid.NewGuid() + ".crt", secondInstanceCertificate);
+        string secondPrivateKeyFilePath = sandbox.CreateFile(Guid.NewGuid() + ".key", secondInstancePrivateKey);
+
+        MemoryFileProvider fileProvider = new();
+
+        fileProvider.IncludeFile(MemoryFileProvider.DefaultAppSettingsFileName,
+            BuildAppSettingsJson(certificateName, firstCertificateFilePath, firstPrivateKeyFilePath));
 
         var configurationBuilder = new ConfigurationBuilder();
-        configurationBuilder.AddCertificate(CertificateName, certificate1FilePath, privateKey1FilePath);
-        IConfigurationRoot configurationRoot = configurationBuilder.Build();
+        configurationBuilder.AddJsonFile(fileProvider, MemoryFileProvider.DefaultAppSettingsFileName, false, true);
```

Comment:
I find this an odd combination. We're using true filesystem changes everywhere else in this test.

Ideally, this test and the one below it are identical, except for the 'act' phase: overwrite appsettings, versus overwrite certificate files.